core/trino-server-rpm/src/main/rpm/preinstall

```diff
@@ -21,9 +33,7 @@ check_if_correct_java_version() {
 # The one argument is the location of java (either $JAVA_HOME or a potential
 # candidate for JAVA_HOME).
   JAVA_VERSION=$(java_version "$1")
-  JAVA_UPDATE=$(echo $JAVA_VERSION | cut -d'_' -f2)
-  JAVA_MAJOR=$(echo $JAVA_VERSION | cut -d'.' -f1)
-  if [[ ("$JAVA_MAJOR" -ge "11") ]]; then
+  if [ $(version_compare "${project.properties.javaversion}" "$JAVA_VERSION") -ge "0" ]; then
```

Comment:
This is nice, but err prone. The version comparison logic can be tricky and i would want to have it as simple as possible. I'd rather keep this in the same spirit it was.

@losipiuk @electrum thoughts?

Reply:
That's why I'm relying on `sort -V`

Reply:
```
     -V, --version-sort
             Sort version numbers.  The input lines are treated as file names in form PREFIX VERSION SUFFIX, where SUFFIX matches the regular expression "(.([A-Za-z~][A-Za-z0-9~]*)?)*".  The files are compared by their prefixes and versions (leading zeros are ignored in version numbers, see example below).  If an input string does not match the pattern, then it is compared using the byte compare function.  All string comparisons are performed in C locale, the locale environment setting is ignored.
```

Reply:
Ya, but how can i know `sort -V` actually sorts _the right_ way from Java perspective?

Reply:
I would keep it as dumb as possible (i.e. not depend on specific semantics of `sort` as we do not have control over what version of that is installed in the system).

Reply:
It should sort correctly according to my reading of https://openjdk.java.net/jeps/223